### DISCUSSION TITLE
Interpreter conflict between '<space>' and ' name

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#115] [Q&A] [answered]
Reading Forth 200x Draft 19.1 I think there is a conflict in 3.4 The Forth text interpreter with respect to the handling of 3.4.1.3 Text interpreter input number conversion. In 3.4.1.3 

     <cnum> := '<char>' 

where `'<char>'` can be any printable ASCII character including space (as defined in 3.1.2.1 Graphic characters).  So in theory I should be able to put a space on the stack or define BL with:

    ok ' '
    ok ' ' CONSTANT BL

but they appear to fail because the manner in which the interpreter is defined it tries to lookup a word ***before*** parsing a number, so single-quote space single-quote is treated like 

    ok ' <word>

where `<word>` is the single-quote.  Should the standard alter the 3.4.1.3 to exclude space from `'<char>'`or modify the interpreter behaviour to try to parse numbers before word lookup?

The reason this came up was that I stumbled on this discussion https://forth-standard.org/proposals/bl-rationale-is-wrong#reply-249 and had a play with my implementation testing stuff and noticed a "bug".  After some reading it appears to be a conflict in the standard draft.

**Comment by ruv** [comment, marked as answer]
You are right, and it's a [known problem](https://forth-standard.org/standard/usage#contribution-33).  Somebody should submit a [proposal](https://forth-standard.org/proposals), and it will be most probably introduced into the next version of the standard.
